Assets/Scripts/InventoryAndItemsScripts/Items/Collectables/InteractCollectable.cs

```diff
@@ -0,0 +1,79 @@
+using Items;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Inventory;
+using Items.Interactables;
+
+namespace Items.Collectables
+{
+    public class InteractCollectable : Collectable, IInteractable
+    {
+        [SerializeField] private int numberOfTimesCanInteract = 1;
+        [SerializeField] private GameObject player;
+        
+        private int numberOfTimesHasInteracted = 0;
+        private bool willDestroyWhenDoneCollecting;
+        private InteractableOverlayHelper interactableOverlay;
+
+        /// <summary>
+        /// Use this property AFTER calling Interact()
+        /// </summary>
+        public bool DestroyAfterInteracting
+        {
+            get
+            {
+                return this.destroyOnCollect;
+            }
+        }
+
+        void Awake()
+        {
+            this.willDestroyWhenDoneCollecting = this.destroyOnCollect;
+            this.destroyOnCollect = false;
+
+            this.interactableOverlay = this.GetComponentInChildren<InteractableOverlayHelper>();
+            if (this.interactableOverlay == null)
```

Comment:
this.interactableOverlay is null is generally better practice, again, probably not necessary